test/Microsoft.Sbom.Tool.Tests/IntegrationTests.cs

```diff
@@ -398,6 +469,67 @@ private static string GetAppName()
         return IsWindows ? "Microsoft.Sbom.Tool.exe" : "Microsoft.Sbom.Tool";
     }
 
+    private void AddExtraFileToManifest(string testFolderPath, string version, object newFile)
+    {
+        var arrayToModify = string.Empty;
+        switch (version)
+        {
+            case "2.2":
+                arrayToModify = "files";
+                break;
+            case "3.0":
+                arrayToModify = "@graph";
+                break;
+            default:
+                return;
+        }
+
+        var manifestFolderPath = AppendFullManifestFolderPath(testFolderPath, spdxVersion: version);
+        var jsonFilePath = Path.Combine(manifestFolderPath, ManifestFileName);
+
+        var jsonContent = File.ReadAllText(jsonFilePath);
+        var jsonDocument = JsonDocument.Parse(jsonContent);
+        var rootElement = jsonDocument.RootElement;
+        var filesArray = rootElement.GetProperty(arrayToModify).EnumerateArray().ToList();
+
+        // Add a new file to manifest.spdx.json
+        filesArray.Add(JsonSerializer.SerializeToElement(newFile));
+
+        // Create a new JSON object with the updated "files" array
```

Comment:
This logic seems complicated if we're basically just trying to modify a object, is there a way to simplify?

Reply:
Simplified in latest commit